lib/Messenger/Handler/AssetUpdateTasksHandler.php

```diff
@@ -62,6 +62,10 @@ private function saveAsset(Asset $asset): void
 
     private function processDocument(Asset\Document $asset): void
     {
+        if ($asset->getMimeType() === 'application/pdf' && $asset->checkIfPdfContainsJS()) {
+            $asset->save(['versionNote' => 'PDF scan result']);
```

Comment:
why call save here? and why create a version vs below in `saveAsset` where version is not created?